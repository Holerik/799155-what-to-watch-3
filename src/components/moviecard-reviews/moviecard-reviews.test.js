// moviecard-reviews.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardReviews from './moviecard-reviews.jsx';

const detailedFilmInfo = [
  {
    id: 0,
    title: `Little Women`,
    poster: `img/little-woman-poster.jpg`,
    altPoster: `ittle Women poster`,
    background: `img/little-wooman-bg.jpg`,
    altBackground: `Movie background`,
    description: `Based on Loiusa May Alcott’s classic novel with the same name, Little Women is a timeless tale of love, life and ambitions`,
    review: `A writer, a sketch artist, an aspiring actress and a budding pianist. All four March sisters are gifted with a unique talent, but they live in the times where the only two choices for women are marriage and death. Writer-director Greta Gerwig`,
    reviews: [1, 2],
    genre: [`Drama`, `Romance`],
    year: 2019,
    duration: `2h 15min`,
    age: `16+`,
    rating: {
      score: `8.0`,
      level: `very good`,
      count: 150,
    },
    director: `Greta Gerwig`,
    starring: [`Saoirse Ronan`, `Emma Watson`, `Florence Pugh`, `Meryl Streep`, `Timothee Chalamet`],
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  },
  {
    id: 1,
    title: `Little Women`,
    poster: `img/little-woman-poster.jpg`,
    altPoster: `ittle Women poster`,
    background: `img/little-wooman-bg.jpg`,
    altBackground: `Movie background`,
    description: `Based on Loiusa May Alcott’s classic novel with the same name, Little Women is a timeless tale of love, life and ambitions`,
    review: `A writer, a sketch artist, an aspiring actress and a budding pianist. All four March sisters are gifted with a unique talent, but they live in the times where the only two choices for women are marriage and death. Writer-director Greta Gerwig`,
    reviews: [1, 2],
    genre: [`Drama`, `Romance`],
    year: 2019,
    duration: `2h 15min`,
    age: `16+`,
    rating: {
      score: `8.0`,
      level: `very good`,
      count: 150,
    },
    director: `Greta Gerwig`,
    starring: [`Saoirse Ronan`, `Emma Watson`, `Florence Pugh`, `Meryl Streep`, `Timothee Chalamet`],
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  },
  {
    id: 2,
    title: `Little Women`,
    poster: `img/little-woman-poster.jpg`,
    altPoster: `ittle Women poster`,
    background: `img/little-wooman-bg.jpg`,
    altBackground: `Movie background`,
    description: `Based on Loiusa May Alcott’s classic novel with the same name, Little Women is a timeless tale of love, life and ambitions`,
    review: `A writer, a sketch artist, an aspiring actress and a budding pianist. All four March sisters are gifted with a unique talent, but they live in the times where the only two choices for women are marriage and death. Writer-director Greta Gerwig`,
    reviews: [1, 2],
    genre: [`Drama`, `Romance`],
    year: 2019,
    duration: `2h 15min`,
    age: `16+`,
    rating: {
      score: `8.0`,
      level: `very good`,
      count: 150,
    },
    director: `Greta Gerwig`,
    starring: [`Saoirse Ronan`, `Emma Watson`, `Florence Pugh`, `Meryl Streep`, `Timothee Chalamet`],
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  },
  {
    id: 3,
    title: `Little Women`,
    poster: `img/little-woman-poster.jpg`,
    altPoster: `ittle Women poster`,
    background: `img/little-wooman-bg.jpg`,
    altBackground: `Movie background`,
    description: `Based on Loiusa May Alcott’s classic novel with the same name, Little Women is a timeless tale of love, life and ambitions`,
    review: `A writer, a sketch artist, an aspiring actress and a budding pianist. All four March sisters are gifted with a unique talent, but they live in the times where the only two choices for women are marriage and death. Writer-director Greta Gerwig`,
    reviews: [1, 2],
    genre: [`Drama`, `Romance`],
    year: 2019,
    duration: `2h 15min`,
    age: `16+`,
    rating: {
      score: `8.0`,
      level: `very good`,
      count: 150,
    },
    director: `Greta Gerwig`,
    starring: [`Saoirse Ronan`, `Emma Watson`, `Florence Pugh`, `Meryl Streep`, `Timothee Chalamet`],
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  },
];

const setMovieCardId = () => {};
const setPageId = () => {};


it(`<MovieCardReviews /> should render movie card with reviews`, () => {
  const tree = renderer
    .create(<MovieCardReviews
      filmsFullInfo={detailedFilmInfo}
      movieId={0}
      setMovieCardId={setMovieCardId}
      setPageId={setPageId}
    >
    </MovieCardReviews>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
