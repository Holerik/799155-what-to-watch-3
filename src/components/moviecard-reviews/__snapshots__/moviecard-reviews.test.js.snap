// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCardReviews /> should render movie card with reviews 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Movie background"
          src="img/little-wooman-bg.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Little Women
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              DramaÂ·Romance
            </span>
            <span
              className="movie-card__year"
            >
              2019
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="add-review.html"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="ittle Women poster"
            height="327"
            src="img/little-woman-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
              onClick={[Function]}
              onMouseOver={[Function]}
            >
              <li
                className="movie-nav__item
                    "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item
                    "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item
                    movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-card__reviews movie-card__row"
          >
            <div
              className="movie-card__reviews-col"
            >
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Bill Goodykoontz
                    </cite>
                    <time
                      className="review__date"
                      dateTime="2015-11-18"
                    >
                      November 18, 2015
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  8,0
                </div>
              </div>
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Amanda Greever
                    </cite>
                    <time
                      className="review__date"
                      dateTime="2015-11-18"
                    >
                      November 18, 2015
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  8,0
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div>
        <div
          className="catalog__movies-list"
        >
          <article
            className="small-movie-card catalog__movies-card"
          >
            <div
              className="small-movie-card__image"
              id="1"
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <img
                alt="ittle Women poster"
                height="175"
                src="img/little-woman-poster.jpg"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
              onClick={[Function]}
            >
              <a
                className="small-movie-card__link"
                href="#"
              >
                Little Women
              </a>
            </h3>
          </article>
          <article
            className="small-movie-card catalog__movies-card"
          >
            <div
              className="small-movie-card__image"
              id="2"
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <img
                alt="ittle Women poster"
                height="175"
                src="img/little-woman-poster.jpg"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
              onClick={[Function]}
            >
              <a
                className="small-movie-card__link"
                href="#"
              >
                Little Women
              </a>
            </h3>
          </article>
          <article
            className="small-movie-card catalog__movies-card"
          >
            <div
              className="small-movie-card__image"
              id="3"
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <img
                alt="ittle Women poster"
                height="175"
                src="img/little-woman-poster.jpg"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
              onClick={[Function]}
            >
              <a
                className="small-movie-card__link"
                href="#"
              >
                Little Women
              </a>
            </h3>
          </article>
        </div>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
